src/compiler/transformers/module/module.ts

```diff
@@ -745,11 +745,20 @@ namespace ts {
                     else if (declaration.kind === SyntaxKind.ImportSpecifier) {
                         const name = (<ImportSpecifier>declaration).propertyName
                             || (<ImportSpecifier>declaration).name;
-                        return createPropertyAccess(
-                            getGeneratedNameForNode(declaration.parent.parent.parent),
-                            getSynthesizedClone(name),
-                            /*location*/ node
-                        );
+                        if (name.originalKeywordKind === SyntaxKind.DefaultKeyword && languageVersion <= ScriptTarget.ES3) {

```

Comment:
I am aware of that. However, my understanding is that or this case which is export/import only "default" will be affected because other reservedWords can't be used as an identifier for function name, variable name , or class name